navigation/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java

```diff
@@ -72,26 +178,45 @@ public void onFailure(Call<DirectionsResponse> call, Throwable throwable) {
     Timber.e(throwable.getMessage());
   }
 
+  @SuppressWarnings({"MissingPermission"})
+  private void initLocationEngine() {
+    locationEngine = new LocationSource(this);
```

Comment:
Can be replaced with the default navigation one, use `MapboxNavigation.getLocationEngine()`